Marlin/src/feature/pause.cpp

```diff
@@ -143,15 +143,16 @@ bool pause_print(const float &retract, const float &z_lift, const float &x_pos,
   // Save current position
   stepper.synchronize();
   COPY(resume_position, current_position);
-
+  
   if (retract) {
     // Initial retract before move to filament change position
-    set_destination_from_current();
+    set_destination_from_current();  // initialise destinations before RunPlan
     destination[E_AXIS] += retract;
     RUNPLAN(PAUSE_PARK_RETRACT_FEEDRATE);
     stepper.synchronize();
+    set_current_from_destination(); // updating current after RunPlan Moves/before DoblockMo
```

Comment:
**RUNPLAN Vs Doblockmove**

> RUNPLAN need ' destinations ' to move and after you manually set ' current pos '

> Doblockmoves apply directly the new current position , and ask planner to go there

For these reason , we are forced to prepare ' current ' and ' destination ' before each
In this case ' retract is a RUNPLAN , then we ensure the current pos and we set destination and because of a doblockmove just after we have to apply the actual current value when finish '

It's a shame ' doblockmove don't implement Extrusion ' , the code could be more clean and more fast , because only one way to calculate a move instead of 2 now , but it's not important , the printer is pause , we don't need high computation

Note ' this is a true ' set_current_destination that works like it says loll '